src/connectors/numeric-menu/connectNumericMenu.ts

```diff
@@ -4,83 +4,88 @@ import {
   isFiniteNumber,
   noop,
 } from '../../lib/utils';
-import { Connector, CreateURL } from '../../types';
+import { Connector, CreateURL, TransformItems } from '../../types';
 import { SearchParameters } from 'algoliasearch-helper';
 
 const withUsage = createDocumentationMessageGenerator({
   name: 'numeric-menu',
   connector: true,
 });
 
-type Option = {
+export type NumericMenuConnectorParamsItem = {
   /**
    * Name of the option
    */
   label: string;
+
   /**
    * Higher bound of the option (<=)
    */
-  start: number;
+  start?: number;
+
   /**
    * Lower bound of the option (>=)
    */
-  end: number;
+  end?: number;
```

Comment:
they  aren't both optional, at least one is required, this means it's more something like:

```ts
NumericMenuConnectorParamsItem = {
  label: string;
  start: string;
  end?: string;} | { label: string; start?: string; end: string }
```

Reply:
@Haroenv Not sure if we should go far on that one. As example, on API Clients we don't go that far on query rules.